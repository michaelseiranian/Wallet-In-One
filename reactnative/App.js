import { useState } from 'react';
import {ThemeProvider} from './src/theme/ThemeProvider'

import 'react-native-gesture-handler';
import Navigation from './Navigation';


import { userContext } from './data';
import {StatusBar} from "expo-status-bar";

export default function App() {
  
  const user = useState({
    'signedIn': false
  });

  return (
    <ThemeProvider>
      <userContext.Provider value={user}>
        <StatusBar barStyle="dark-content" />
        <Navigation></Navigation>
      </userContext.Provider>
    </ThemeProvider>
  );
}
