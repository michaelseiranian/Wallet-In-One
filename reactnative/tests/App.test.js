import {it} from "@jest/globals";
import {render} from "@testing-library/react-native";
import App from "../App";
import expect from "expect";
import * as React from 'react';

const mockNavigation = jest.fn();
jest.mock('@react-navigation/native', () => ({ ...jest.requireActual('@react-navigation/native'), useNavigation: () => { return mockNavigation; }, }));

it('renders correctly', () => {
  const snapshot = render(<App />);
  expect(snapshot).toMatchSnapshot();
});