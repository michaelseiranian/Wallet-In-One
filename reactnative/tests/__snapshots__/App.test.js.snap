// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              [
                {
                  "display": "flex",
                  "overflow": undefined,
                },
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              {
                "close": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": {
                  "animation": "spring",
                  "config": {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  {
                    "flex": 1,
                    "transform": [
                      {
                        "translateX": 0,
                      },
                      {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="box-none"
                  style={
                    [
                      {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      [
                        {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityIgnoresInvertColors={true}
                        style={
                          {
                            "height": "100%",
                            "width": "100%",
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            [
                              {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              {
                                "height": "100%",
                                "width": "100%",
                              },
                              undefined,
                            ]
                          }
                        />
                        <View>
                          <Image
                            resizeMode="contain"
                            source={1}
                            style={
                              {
                                "height": 280,
                                "marginLeft": "15%",
                                "marginTop": "-15%",
                                "width": 280,
                              }
                            }
                          />
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                {
                                  "alignSelf": "center",
                                  "backgroundColor": "black",
                                  "borderRadius": 25,
                                  "color": "purple",
                                  "fontSize": 17,
                                  "fontWeight": "bold",
                                  "marginTop": "-10%",
                                  "padding": "2%",
                                  "textAlign": "center",
                                  "width": "30%",
                                }
                              }
                            >
                              About Us
                            </Text>
                          </View>
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                {
                                  "backgroundColor": "purple",
                                  "borderRadius": 25,
                                  "color": "white",
                                  "fontSize": 27,
                                  "fontWeight": "bold",
                                  "marginLeft": "11%",
                                  "marginTop": "65%",
                                  "padding": "2%",
                                  "textAlign": "center",
                                  "width": "75%",
                                }
                              }
                            >
                              Sign Up
                            </Text>
                          </View>
                          <View
                            accessibilityState={
                              {
                                "busy": undefined,
                                "checked": undefined,
                                "disabled": undefined,
                                "expanded": undefined,
                                "selected": undefined,
                              }
                            }
                            accessibilityValue={
                              {
                                "max": undefined,
                                "min": undefined,
                                "now": undefined,
                                "text": undefined,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              {
                                "opacity": 1,
                              }
                            }
                          >
                            <Text
                              style={
                                {
                                  "backgroundColor": "white",
                                  "borderRadius": 25,
                                  "color": "purple",
                                  "fontSize": 27,
                                  "fontWeight": "bold",
                                  "marginLeft": "11%",
                                  "marginTop": "5%",
                                  "padding": "2%",
                                  "textAlign": "center",
                                  "width": "75%",
                                }
                              }
                            >
                              Log In
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
